Add support for case sensitive schema name

Praveen2112 opened the pull request.

Praveen2112 committed:
Introduce NameCanonicalizer

Praveen2112 committed:
Remove lower case conversion for objects for schema/table/view name

 These will be enforced by the query engine and by the connector.

Praveen2112 committed:
Add support for case sensitive schema name

Supports create/rename/drop schema. But using it as a part of table name or column name is restricted.
However we can specify it via `USE` or specify them in session

Praveen2112 committed:
Allow BaseConnectorTest to create case sensitive schema

Praveen2112 committed:
Introduce canonicalize for JDBC client

Praveen2112 committed:
Add support for case sensitive schema name in Postgres connector

kokosing commented on core/trino-spi/src/main/java/io/trino/spi/connector/ConnectorMetadata.java:
```diff
@@ -1145,4 +1146,14 @@ default MaterializedViewFreshness getMaterializedViewFreshness(ConnectorSession
+     * Canonicalizes the provided SQL identifier according to connector-specific rules
+     * for the purpose of providing the name in metadata APIs
+     */
+    default String canonicalize(ConnectorSession session, String identifier, boolean delimited)
```
Why do we pass session?

Praveen2112 replied:
If some connector wants to maintain the legacy way of matching ... then they can use this to toggle between the legacy and actual way.

lukasz-walkiewicz committed:
Configure SSL for Hydra IdP in product tests

sumannewton committed:
Fix incorrect pushdown if IS NULL predicate in Elasticsearch

findepi committed:
Encode unbounded varchar value without a CAST

findinpath committed:
Document luhn_check function

SamWheating committed:
Fix typo in Iceberg connector documentation

findinpath committed:
Remove trailing spaces

findepi committed:
Exclude snakeyaml from the build

We reference an version of the library. Removing on the assumption it's
not actually needed.

guyco33 committed:
Bump mysql-connector-java to 8.0.22

electrum committed:
Remove usage of /v1/info for JDBC driver

findinpath committed:
Add support for generic parameters in `make_set_digest` function

A widely applied usecase  for minhash algorithm is to check the
similarity of two texts.
Therefore the current implementation adds support for creating a
`setdigest` aggregation for (among other types) varchar slices.

sopel39 committed:
Add assertEventually overload with retryFrequency

mosabua committed:
Add Redshift requirements

findepi committed:
Remove redundant else

findepi committed:
Simplify assignment

findepi committed:
Propagate runtime errors in ExpressionInterpreter

The previous catching of `RuntimeException` could mask implementation issues such as
`VerifyException` or `NullPointerException`.

findepi committed:
Simplify code

- process Optional fluently
- group variable initialization

mosabua committed:
Ignore VSCode config folder

tooptoop4 committed:
Update esri-geometry-api to 2.2.4
